stock_picking_group_by_partner_by_carrier/report/report_delivery_slip.xml

```diff
@@ -0,0 +1,176 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <template
+        id="report_delivery_document"
+        inherit_id="stock.report_delivery_document"
+        priority="120"
+    >
+        <div class="page" position="before">
+            <t t-set="report_lines" t-value="o.get_delivery_report_lines()" />
+        </div>
+        <!-- overrides when the picking is not done -->
+        <xpath
+            expr='//table[@name="stock_move_table"]/tbody/t[@t-set="lines"]'
+            position="attributes"
+        >
+            <attribute name="t-value">report_lines</attribute>
+        </xpath>
+
+        <xpath
+            expr='//table[@name="stock_move_table"]/tbody/tr//span[@t-field="move.product_id"]/..'
+            position="replace"
```

Comment:
@TDU @jbaudoux I don't like replaces but I haven't found a better method at this time

Reply:
hide it setting attribute `t-if=False` then add your later